R/ResampleInstances.R

```diff
@@ -67,3 +67,63 @@ instantiateResampleInstance.SpRepCVDesc = function(desc, size, task = NULL, coor
   g = as.factor(rep(seq_len(desc$reps), each = folds))
   makeResampleInstanceInternal(desc, size, train.inds = train.inds, test.inds = test.inds, group = g)
 }
+
+
+instantiateResampleInstance.FixedWindowCVDesc = function(desc, size, task = NULL, coords) {
+  initial.window.abs = floor(desc$initial.window * size)
+  assertInt(initial.window.abs, lower = 1)
+  if (size - initial.window.abs < desc$horizon) {
+    stop(catf("The initial window is %i observations while the data is %i observations. \n There is not enough data left (%i observations) to create a test set for a %i size horizon.",
+              initial.window.abs, size, initial.window.abs - size, desc$horizon))
+  }
+  skip = desc$skip
+  stops  = (seq(size))[initial.window.abs:(size - desc$horizon)]
+  starts = stops - initial.window.abs + 1
+  train.inds = mapply(seq, starts, stops, SIMPLIFY = FALSE)
+  test.inds  = mapply(seq, stops + 1, stops + desc$horizon, SIMPLIFY = FALSE)
+
+  if (skip > 0) {
+    train.inds = thin(train.inds, skip = skip)
+    test.inds = thin(test.inds, skip = skip)
+  }
+
+  if (length(test.inds) == 0)
+    stop("Skip is too large and has removed all resampling instances. Please lower the value of skip.")
+
+  # If the last value is not included, shift everything over by one
+  if (test.inds[[length(test.inds)]][desc$horizon] != size) {
+    warning("The last observation was excluded. To include the final observation please change either the initial.window or horizon.")
```

Comment:
So the purpose of this check is to make sure that the user isn't accidentally excluding data, right? Then I would make it more general and check for any number of excluded observations at the end, in particular as excluding e.g. 3 observations at the end is worse than 1.

Reply:
yes, now it checks how much data is missing and includes it in the error